File: JULY-2025/30-07-2025/p2419.cs
public class Solution {
    public int LongestSubarray(int[] nums) {
        int max = 0; // highest nums value found so far
        int count = 0; // running count of max nums currently found
        int precord = 1; // previous record (will be at least 1)

        for (int i=0; i<nums.Length; i++)
        {
            if (nums[i] < max) // sqeuence ended/invalid
            {                
                count = 0;
            }

            if (nums[i] == max) // best candidate, start/keep counting
            {
                count++;
                if (count > precord) precord = count; // also check for record setting
            }

            if (nums[i] > max) // new max
            {
                max = nums[i];
                count = 1;
                precord = 1;
            }
        }
        
        return precord;
    }
}

File: JULY-2025/30-07-2025/p2419.rs
impl Solution {
    pub fn longest_subarray(nums: Vec<i32>) -> i32 {
        let n=nums.len();
        let mut maxEl=nums[0].clone();
        let mut cnt=0;
        let mut mCnt=0;
        for i in 0..n {
            if (maxEl<nums[i]) {
                maxEl=nums[i];
                cnt=0;
                mCnt=0;
            }
            if (maxEl==nums[i]) {
                cnt+=1;
                mCnt= mCnt.max(cnt);
            }else{
                cnt=0;
            }

        } mCnt
    }
}